Core/Src/DataReadingTask.c

```diff
@@ -1,5 +1,76 @@
 #include "Tasks.h"
+#include "IMU.h"
+#include "GPS.h"
+#include "CANBus.h"
+#include "SDCard.h"
+#include "radio.h"
+
+//This function reads from IMU, CAN, and GPS data structures, copies their current values, and sends them both through ethernet 
+// (and to be saved on SD card when SD card code is finished).
 
 void DataReadingTask(void* argument){
-    
+    while(1) {
+        CANMSG_t CAN_Data;
+        IMUData_t IMU_Data;
+        GPSData_t GPS_Data;
+        SDCard_t SDCard_data;
+
+        EthernetMSG_t Message;
+        //BaseType_t status;
+        BaseType_t CANBus_status;
+        BaseType_t GPS_status;
+        //BaseType_t SDCard_status;
+        HAL_StatusTypeDef error;
+
+        // Retrieve GPS data
+        GPS_status = GPS_ReadData(&GPS_Data);
+        // Assign SD card timestamp
+        memcpy(SDCard_data.time, GPS_Data.time, sizeof(GPS_Data.time));
+
+        //Send GPS data and log in SD card
+        if (GPS_status != pdFALSE) {
+            Message.id= GPS;
+            Message.length = sizeof(GPS_Data);
+            memcpy(&Message.data.GPSData, &GPS_Data, sizeof(GPS_Data));
+            Ethernet_PutInQueue(&Message);// TODO: Add error handling on Ethernet_PutInQueue()
+
+            SDCard_data.id = GPS_SDCard;
+            memcpy(&SDCard_data.data.GPSData, &GPS_Data, sizeof(GPS_Data));
+            SDCard_PutInQueue(&SDCard_data); // TODO: Add error handling 
+            
+
+        } else {} // TODO: error handling for GPS_ReadData()
+
+        // Send IMU data and log in SD card
+        error = IMU_GetMeasurements(&IMU_Data);
+        if (error != HAL_ERROR){
```

Comment:
You don't need the extra variable error here. Try getting rid of redundant variables.